### ISSUE TITLE
Alternator charging problem 

### ISSUE THREAD

**Issue by BeerMoneyMotorsports** [rusefi/hellen-NB1-issues#29]
**Board**- NB1 rev B
**Log**- https://rusefi.com/online/view.php?log=640

Problem-  Alternator will stop charging and voltage will drop to ~12v.  It will stay there a few seconds and then it will ramp up to 17-18v before coming back to normal 14.2v.  

The log indicates that the ECU is indeed requesting the activation of the alternator but it is not responding.  

I have changed the alternator, reviewed chassis wiring, tried another Hellen.  No improvements.  I have also tried several settings in the tune to correct with no improvements.  

I **will be testing alternator output wire with an Oscilloscope today.**   I started by bench testing with the scope.  ecu had 12v supply and was stimulated to simulate 1200rpm.
[alternator bench test.pdf](https://github.com/rusefi/hellen-NB1-issues/files/8112146/alternator.bench.test.pdf)

**Comment by BeerMoneyMotorsports** [comment]
[alternator idle.pdf](https://github.com/rusefi/hellen-NB1-issues/files/8112606/alternator.idle.pdf)
[alternator idle high load.pdf](https://github.com/rusefi/hellen-NB1-issues/files/8112609/alternator.idle.high.load.pdf)
[alt reved 12v.pdf](https://github.com/rusefi/hellen-NB1-issues/files/8112612/alt.rev.12v.pdf)
[alternator 17v.pdf](https://github.com/rusefi/hellen-NB1-issues/files/8112617/alternator.17v.pdf)

**Comment by BeerMoneyMotorsports** [comment]
circuit tests
Alternator control(1o) wire to ground
ECU disconnected- 1.026 kohm
ECU and alternator disconnected- OL

**Comment by BeerMoneyMotorsports** [comment]
[Matthew Kennedy](https://app.slack.com/team/U80K9BYVD)  [7:36 PM](https://rusefi.slack.com/archives/C0323M3MPU4/p1645490212114379)
can you measure alt pin -> ground on the ECU?
[7:36](https://rusefi.slack.com/archives/C0323M3MPU4/p1645490215172999)
unplugged from car
[7:38](https://rusefi.slack.com/archives/C0323M3MPU4/p1645490282592919)
are either of R309 or R310 installed?
[7:38](https://rusefi.slack.com/archives/C0323M3MPU4/p1645490310461799)
I have a hunch neither is installed

[BeerMoneyMotorsports](https://app.slack.com/team/U01JUH5H6EN):speech_balloon:  [7:38 PM](https://rusefi.slack.com/archives/C0323M3MPU4/p1645490314242429)
OL
[7:43](https://rusefi.slack.com/archives/C0323M3MPU4/p1645490599050929)
I believe r310 is in place.
[7:44](https://rusefi.slack.com/archives/C0323M3MPU4/p1645490691092599)
I dont see r309 on ibom

[Matthew Kennedy](https://app.slack.com/team/U80K9BYVD)  [7:45 PM](https://rusefi.slack.com/archives/C0323M3MPU4/p1645490724637339)
OK, I expect R310 to be present and R309 missing
[7:45](https://rusefi.slack.com/archives/C0323M3MPU4/p1645490730800099)
can you measure R310?
[7:45](https://rusefi.slack.com/archives/C0323M3MPU4/p1645490732764369)
should be 0

[BeerMoneyMotorsports](https://app.slack.com/team/U01JUH5H6EN):speech_balloon:  [7:48 PM](https://rusefi.slack.com/archives/C0323M3MPU4/p1645490934363359)
Give me 15

[Matthew Kennedy](https://app.slack.com/team/U80K9BYVD)  [7:49 PM](https://rusefi.slack.com/archives/C0323M3MPU4/p1645490965979339)
and check the voltage both sides of R310 with ECU connected/ign on/whatever

[BeerMoneyMotorsports](https://app.slack.com/team/U01JUH5H6EN):speech_balloon:  [8:03 PM](https://rusefi.slack.com/archives/C0323M3MPU4/p1645491838119039)
can i do that on the bench?  its a royal pain in the ass in the car
[8:04](https://rusefi.slack.com/archives/C0323M3MPU4/p1645491858017629)
i dont have an ibom for nb1-b
[8:04](https://rusefi.slack.com/archives/C0323M3MPU4/p1645491873772269)
i need some education on how to get that
[8:04](https://rusefi.slack.com/archives/C0323M3MPU4/p1645491892349969)
its not on https://rusefi.com/docs/ibom/

[andrey](https://app.slack.com/team/U4PVDN5L7)  [8:07 PM](https://rusefi.slack.com/archives/C0323M3MPU4/p1645492031597889)
https://github.com/rusefi/hellen-NB1/tree/main/boards/hellen64_NB1-a/board has ibom
New

[Matthew Kennedy](https://app.slack.com/team/U80K9BYVD)  [8:07 PM](https://rusefi.slack.com/archives/C0323M3MPU4/p1645492052666479)
Bench OK if power supply the same as in a car

[andrey](https://app.slack.com/team/U4PVDN5L7)  [8:07 PM](https://rusefi.slack.com/archives/C0323M3MPU4/p1645492054357519)
https://github.com/rusefi/hellen-NB1/tree/main/boards/hellen64_NB1-b/board does not
[8:07](https://rusefi.slack.com/archives/C0323M3MPU4/p1645492062693989)
weird worth poking [@andreika](https://rusefi.slack.com/team/U4YULLW6Q)

[BeerMoneyMotorsports](https://app.slack.com/team/U01JUH5H6EN):speech_balloon:  [8:08 PM](https://rusefi.slack.com/archives/C0323M3MPU4/p1645492113680859)
that will have to be tomorrow then.  that part of the board had changes from A to B
[8:11](https://rusefi.slack.com/archives/C0323M3MPU4/p1645492290519869)
[@Matthew Kennedy](https://rusefi.slack.com/team/U80K9BYVD) r310 shows vbatt on both sides
[8:11](https://rusefi.slack.com/archives/C0323M3MPU4/p1645492312799999)
warning, i am assuming this is r310.  things moved around from A to B
[8:12](https://rusefi.slack.com/archives/C0323M3MPU4/p1645492331011519)
it isnt in the exact same spot but very close

**Comment by BeerMoneyMotorsports** [comment]
awaiting ibom for confirmation of location of r309 and r310.  Poked @andreika-git for support with ibom for NB1 rev-B

**Comment by BeerMoneyMotorsports** [comment]
@mck1117 please advise what the expected voltage output should be on the oscilloscope readout.

**Comment by mck1117** [comment]
I would expect to see:

1. 12v on both sides of R310
2. 12v on pin 6 of U302
![image](https://user-images.githubusercontent.com/568254/155049959-0cb3baed-1046-4424-908e-e1a66222b762.png)
3. PWM at 0v/12v on U302 pin 7 and on the alternator output pin.

**Comment by andreika-git** [comment]
![image](https://user-images.githubusercontent.com/15104768/155073916-e0af7ad1-eabf-414c-bac3-bc5c779de8b8.png)
R310 is the one with "12" label.

**Comment by BeerMoneyMotorsports** [comment]
![D36AE490-04B6-4DE9-AEAF-86775C0D1D25](https://user-images.githubusercontent.com/82395131/155134050-7810771b-4b76-425a-955d-01e3b9a0cff7.png)

**Comment by BeerMoneyMotorsports** [comment]
Pin 6 u302= 12v
Pin 7 u302= 12v pwn(on bench)

is there a good way to simulate it being on the car?  Connecting alternator output to some load maybe?

**Comment by BeerMoneyMotorsports** [comment]
hellen nb1 rev A shows different performance. No charging issues.


[nb1_a_alternator.pdf](https://github.com/rusefi/hellen-NB1-issues/files/8164810/nb1_a_alternator.pdf)

**Comment by andreika-git** [comment]
The main difference between rev.A and B is this:
![image](https://user-images.githubusercontent.com/15104768/156251413-eb6f46f8-57a7-4c8f-95f9-d20851e91592.png)
Maybe try to replace R23 with a jumper (0 ohms resistor)?

**Comment by mck1117** [comment]
![91D93365-6F62-4E51-B396-BCE713DD0DE6](https://user-images.githubusercontent.com/568254/156254113-91eca043-5d98-4157-9f7f-43052d82e402.jpeg)

Rev A, alternator disconnected. This looks like the tc4427 doesn't have power at all.

**Comment by BeerMoneyMotorsports** [comment]
@andreika-git there were also changes to the output module?  added some circuit protections s well?

**Comment by andreika-git** [comment]
> @andreika-git there were also changes to the output module? added some circuit protections s well?

Yes, we've added more protection around TC4427:
output-0.3 (Now):
![image](https://user-images.githubusercontent.com/15104768/156318049-64d2477a-b370-4358-b143-dde07bda1451.png)

output-0.2 (Before):
![image](https://user-images.githubusercontent.com/15104768/156318124-511a2649-9359-4459-8280-5b67bab34be4.png)
![image](https://user-images.githubusercontent.com/15104768/156318155-662d7e7c-1649-4f18-8d28-cc345b5225e7.png)

**Comment by BeerMoneyMotorsports** [comment]
@mck1117 @andreika-git I have retested NB1 rev A.  Previous test that showed only ~2v was due to user error.  rev B is still only ~2v.

![image](https://user-images.githubusercontent.com/82395131/156462164-a441248b-b0a4-4ae7-ab39-1d37aa7f9311.png)

**Comment by BeerMoneyMotorsports** [comment]
i was able to probe the 4227 while on the car.  i found that at pin 7(output), i was getting a nice battery voltage pwm signal.  Exactly as expected.  I then bypassed R23(1k) and R312(100) and retested.  we then had 11.1v pwm and great alternator control.    It was determined that the alternator resistance must change during operation.  I removed R23 and left R312.  ~4.5v pwm and good alternator control.  so the problem was a result of the double resistors on the output.  We need to replace R23 with a 0ohm resistor or delete from circuit.  i am creating tickets for it.  this can be closed.
